pkg/streamingpromql/query.go

```diff
@@ -81,12 +81,17 @@ func (e *Engine) newQuery(ctx context.Context, queryable storage.Queryable, opts
 		return nil, fmt.Errorf("could not get memory consumption limit for query: %w", err)
 	}
 
+	memoryConsumptionTracker := limiter.NewMemoryConsumptionTracker(maxEstimatedMemoryConsumptionPerQuery, e.queriesRejectedDueToPeakMemoryConsumption)
+	stats, err := types.NewQueryStats(timeRange, e.enablePerStepStats && opts.EnablePerStepStats(), memoryConsumptionTracker)
+	if err != nil {
+		return nil, fmt.Errorf("could not get create stats for query: %w", err)
```

Comment:
Same comment as in `stats.go`: there's no need to wrap the error here, and it might lead to a confusing error message for the user as well.

Reply:
I have two questions:
1. We are wrapping error just above:
```
maxEstimatedMemoryConsumptionPerQuery, err := e.limitsProvider.GetMaxEstimatedMemoryConsumptionPerQuery(ctx)
if err != nil {
	return nil, fmt.Errorf("could not get memory consumption limit for query: %w", err)
}

Should we "unwrap" it then?
2. Does it make sense to log something in this case? Right now, we only fail to create QueryStats if the estimated memory is exceeded. Should we log an error when that happens to make it easier to debug potential query failures due to high memory usage?
```

Reply:
> 1. We are wrapping error just above:

This is a different scenario: in this case, we failed to retrieve the memory consumption limit for the query, which should never happen in practice. So wrapping the error here is fine and helpful, as we can't guarantee the returned error has enough context about the problem.

> 2. Does it make sense to log something in this case?

We haven't done this elsewhere. I'd be very surprised if a query failed at this point due to the memory consumption limit - this is typically set to 1 GB and so the query would need to be for a large number of steps to hit the limit here.